### PR TITLE
[nlir][vector] Constrain `ContractionOpToMatmulOpLowering`

### PR DESCRIPTION
**Pull request by banach-space**
Disables `ContractionOpToMatmulOpLowering` for scalable vectors. This
pattern is meant to enable lowering to `llvm.matrix.multiply` - I'm not
aware of any use of that in the context of scalable vectors.

### CODE REVIEW

File: mlir/lib/Dialect/Vector/Transforms/LowerVectorContract.cpp [hunk 1]
```diff
@@ -1302,13 +1304,18 @@ FailureOr<Value> ContractionOpToMatmulOpLowering::matchAndRewriteMaskableOp(
       !isReductionIterator(iteratorTypes[2]))
     return failure();
 
+  Type opResType = op.getType();
+  VectorType vecType = dyn_cast<VectorType>(opResType);
+  if (vecType && vecType.isScalable()) {
+    // This should be sufficient to reject all cases with scalable vectors.
```
Comment (c-rhodes) [inline mlir/lib/Dialect/Vector/Transforms/LowerVectorContract.cpp:1310]: nit: is the ambiguity is necessary, surely this is sufficient to reject all cases with scalable vectors?

**Review by c-rhodes — APPROVED** [review]
Left one minor comment but otherwise LGTM cheers. I do have a wider question tho, should `vector.matrix_multiply` be disabled for scalable vectors at the op-level?

**Comment by banach-space** [comment]
> I do have a wider question tho, should `vector.matrix_multiply` be disabled for scalable vectors at the op-level?

+1 I'll do that in a follow-up.

**Comment by banach-space** [comment]
> [nlir] 🥲

Muscle memory failing me :(

**Pull request merged**